inject/src/main/java/io/micronaut/context/DefaultApplicationContext.java

```diff
@@ -120,11 +121,20 @@ public DefaultApplicationContext(@NonNull ApplicationContextConfiguration config
     }
 
     @Override
-    protected @NonNull List<BeanDefinitionReference> resolveBeanDefinitionReferences() {
-        if (resolvedBeanReferences != null) {
+    protected @NonNull List<BeanDefinitionReference> resolveBeanDefinitionReferences(Predicate<BeanDefinitionReference> predicate) {
```

Comment:
Can we retain the original signature and delegate just in case to avoid the breaking change